plugins/community/scalapb-scala/v0.11.12/Dockerfile

```diff
@@ -0,0 +1,16 @@
+# syntax=docker/dockerfile:1.4
+FROM debian:bullseye-20221114 AS build
+
+ARG TARGETARCH
+
+RUN apt-get update \
+ && apt-get install -y curl
+
+RUN curl -fsSL -o protoc-gen-scala.sh https://repo1.maven.org/maven2/com/thesamet/scalapb/protoc-gen-scala/0.11.12/protoc-gen-scala-0.11.12-unix.sh \
+    && chmod +x protoc-gen-scala.sh
+
+FROM gcr.io/distroless/java11-debian11
+COPY --from=build /bin/sh /bin/sh
```

Comment:
The distroless images can get a busybox shell by using the `:debug` Docker tag, but I'd prefer if we could somehow just get a .jar file and execute that and not require a shell.

Reply:
I did some more testing - we can simply just rename the .sh file to .jar and it works without a shell:

```diff
diff --git a/plugins/community/scalapb-scala/v0.11.12/Dockerfile b/plugins/community/scalapb-scala/v0.11.12/Dockerfile
index 333ff33..645dd7f 100644
--- a/plugins/community/scalapb-scala/v0.11.12/Dockerfile
+++ b/plugins/community/scalapb-scala/v0.11.12/Dockerfile
@@ -6,11 +6,9 @@ ARG TARGETARCH
 RUN apt-get update \
  && apt-get install -y curl

-RUN curl -fsSL -o protoc-gen-scala.sh https://repo1.maven.org/maven2/com/thesamet/scalapb/protoc-gen-scala/0.11.12/protoc-gen-scala-0.11.12-unix.sh \
-    && chmod +x protoc-gen-scala.sh
+RUN curl -fsSL -o protoc-gen-scala.jar https://repo1.maven.org/maven2/com/thesamet/scalapb/protoc-gen-scala/0.11.12/protoc-gen-scala-0.11.12-unix.sh

 FROM gcr.io/distroless/java11-debian11
-COPY --from=build /bin/sh /bin/sh
-COPY --from=build --link /protoc-gen-scala.sh .
+COPY --from=build --link /protoc-gen-scala.jar .
 USER nobody
-ENTRYPOINT [ "/bin/sh", "/protoc-gen-scala.sh" ]
+ENTRYPOINT [ "/usr/bin/java", "-jar", "/protoc-gen-scala.jar" ]
```

Reply:
The process for doing so is a bit more convoluted, additionally this is how the maintainer of the project distributes the artifacts. It seems safer to me to use that instead of building it from scratch. As there is no guarantee that internal build processes won't change in subsequent versions and we know that this is correct.

I didn't opt for debug because as I understand it it includes a lot more than just a shell

Reply:
This doesn't change to require building it differently - it is using the artifact as-is.

If you look at the start of the shell script, it contains the following lines:
```
#!/usr/bin/env sh
@ 2>/dev/null # 2>nul & echo off & goto BOF
:
exec java -jar $JAVA_OPTS "$0" "$@"
exit

:BOF
java -jar %JAVA_OPTS% "%~dpnx0" %*
exit /B %errorlevel%

... # Embedded jar file
```

So this is just a small wrapper to re-invoke it as `java -jar protoc-gen-scala.sh` for unix platforms and a different invocation for Windows. The key thing is that you can pass the shell script directly to `java -jar` and it works without shipping a separate shell.

Reply:
Sorry comment was for your original post, github didn't tell me there was a new message, will make the change per your testing

Reply:
Good find!